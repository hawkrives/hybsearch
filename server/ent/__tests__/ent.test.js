/* eslint-env jest */
require('jest-specific-snapshot')

const { strictSearch: search, formatData } = require('../index')
const { parse: newickToJson } = require('../../newick')
const {pruneOutliers} = require('../../lib/prune-newick')
const fs = require('fs')
const path = require('path')
const cloneDeep = require('lodash/cloneDeep')

const base = path.join(__dirname, '..', '..', '..', 'data', '__supporting__')
const blacklist = ['bataguridae-cytb.gb', 'trionychcytb.gb']
const files = fs
	.readdirSync(base)
	.filter(name => !name.startsWith('.') && !name.startsWith('_'))
	.filter(name => name.endsWith('.gb'))
	.filter(name => !blacklist.includes(name))

for (const file of files) {
	const read = name => fs.readFileSync(path.join(base, file, name), 'utf-8')

	const fasta = read('aligned-fasta')
	const content = read('newick-tree')
	const _tree = newickToJson(content)

	test(file + ' (pruned)', () => {
		let tree = cloneDeep(_tree)
		let { prunedNewick } = pruneOutliers(tree, fasta)
		let actual = formatData(search(prunedNewick, fasta))
		expect(actual.split('\n').filter(str => str.length > 0)).toMatchSpecificSnapshot(path.join('.', '__snapshots__', file + '.pruned.hybsnap'))
	})

	test(file + ' (unpruned)', () => {
		let tree = cloneDeep(_tree)
		let actual = formatData(search(tree, fasta))
		expect(actual.split('\n').filter(str => str.length > 0)).toMatchSpecificSnapshot(path.join('.', '__snapshots__', file + '.hybsnap'))
	})
}
