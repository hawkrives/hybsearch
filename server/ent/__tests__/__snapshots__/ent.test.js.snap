// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coyote-test 1`] = `
Array [
  "Canis_latrans (EU400577) / Canis_latrans (EU400577)",
  "Canis_latrans (EU400578) / Canis_latrans (EU400578)",
]
`;

exports[`emydura 1`] = `
Array [
  "Emydura subglobosa (KC692462) / Emydura subglobosa (KC692462)",
  "Emydura subglobosa (NC_026048) / Emydura subglobosa (NC_026048)",
  "Emydura subglobosa (KC755184) / Emydura subglobosa (KC755184)",
]
`;

exports[`emydura-short 1`] = `
Array [
  "Emydura subglobosa (KC755184) / Emydura subglobosa (KC755184)",
]
`;

exports[`kino 1`] = `
Array [
  "",
]
`;

exports[`plaamyd 1`] = `
Array [
  "Amyda cartilaginea (LM537480) / Amyda cartilaginea (LM537480)",
  "Amyda cartilaginea (LM537481) / Amyda cartilaginea (LM537481)",
  "Amyda cartilaginea (LM537485) / Amyda cartilaginea (LM537485)",
  "Amyda cartilaginea (LM537484) / Amyda cartilaginea (LM537484)",
  "Amyda cartilaginea (LM537483) / Amyda cartilaginea (LM537483)",
  "Amyda cartilaginea (LM537464) / Amyda cartilaginea (LM537464)",
  "Amyda cartilaginea (LM537463) / Amyda cartilaginea (LM537463)",
  "Amyda cartilaginea (AY259550) / Amyda cartilaginea (AY259550)",
  "Amyda cartilaginea (JN232534) / Amyda cartilaginea (JN232534)",
  "Amyda cartilaginea (LM537465) / Amyda cartilaginea (LM537465)",
  "Amyda cartilaginea (LM537467) / Amyda cartilaginea (LM537467)",
  "Amyda cartilaginea (LM537491) / Amyda cartilaginea (LM537491)",
  "Amyda cartilaginea (LM537492) / Amyda cartilaginea (LM537492)",
  "Amyda cartilaginea (LM537476) / Amyda cartilaginea (LM537476)",
  "Amyda cartilaginea (LM537493) / Amyda cartilaginea (LM537493)",
  "Amyda cartilaginea (LM537494) / Amyda cartilaginea (LM537494)",
]
`;

exports[`smallerlepus 1`] = `
Array [
  "",
]
`;

exports[`trio171sequence 1`] = `
Array [
  "Amyda cartilaginea (LM537480) / Amyda cartilaginea (LM537480)",
  "Amyda cartilaginea (LM537481) / Amyda cartilaginea (LM537481)",
  "Amyda cartilaginea (LM537485) / Amyda cartilaginea (LM537485)",
  "Amyda cartilaginea (LM537484) / Amyda cartilaginea (LM537484)",
  "Amyda cartilaginea (LM537483) / Amyda cartilaginea (LM537483)",
  "Amyda cartilaginea (LM537464) / Amyda cartilaginea (LM537464)",
  "Amyda cartilaginea (LM537463) / Amyda cartilaginea (LM537463)",
  "Amyda cartilaginea (AY259550) / Amyda cartilaginea (AY259550)",
  "Amyda cartilaginea (JN232534) / Amyda cartilaginea (JN232534)",
  "Amyda cartilaginea (LM537465) / Amyda cartilaginea (LM537465)",
  "Amyda cartilaginea (LM537467) / Amyda cartilaginea (LM537467)",
  "Amyda cartilaginea (LM537491) / Amyda cartilaginea (LM537491)",
  "Amyda cartilaginea (LM537492) / Amyda cartilaginea (LM537492)",
  "Amyda cartilaginea (LM537476) / Amyda cartilaginea (LM537476)",
  "Amyda cartilaginea (LM537493) / Amyda cartilaginea (LM537493)",
  "Amyda cartilaginea (LM537494) / Amyda cartilaginea (LM537494)",
]
`;

exports[`trionychcytb 1`] = `
Array [
  "Amyda cartilaginea (LM537481) / Amyda cartilaginea (LM537481)",
  "Amyda cartilaginea (LM537480) / Amyda cartilaginea (LM537480)",
  "Amyda cartilaginea (LM537463) / Amyda cartilaginea (LM537463)",
  "Amyda cartilaginea (LM537464) / Amyda cartilaginea (LM537464)",
  "Amyda cartilaginea (LM537483) / Amyda cartilaginea (LM537483)",
  "Amyda cartilaginea (LM537484) / Amyda cartilaginea (LM537484)",
  "Amyda cartilaginea (LM537485) / Amyda cartilaginea (LM537485)",
  "Amyda cartilaginea (LM537467) / Amyda cartilaginea (LM537467)",
  "Amyda cartilaginea (LM537465) / Amyda cartilaginea (LM537465)",
  "Amyda cartilaginea (JN232534) / Amyda cartilaginea (JN232534)",
  "Amyda cartilaginea (AY259550) / Amyda cartilaginea (AY259550)",
  "Amyda cartilaginea (LM537492) / Amyda cartilaginea (LM537492)",
  "Amyda cartilaginea (LM537491) / Amyda cartilaginea (LM537491)",
  "Amyda cartilaginea (LM537494) / Amyda cartilaginea (LM537494)",
  "Amyda cartilaginea (LM537493) / Amyda cartilaginea (LM537493)",
  "Amyda cartilaginea (LM537476) / Amyda cartilaginea (LM537476)",
  "Palea steindachneri (AY743417) / Palea steindachneri (AY743417)",
]
`;

exports[`vulpes 1`] = `
Array [
  "Vulpes_vulpes (KU378531) / Vulpes_vulpes (KU378531)",
  "Vulpes_vulpes (KU378550) / Vulpes_vulpes (KU378550)",
  "Vulpes_vulpes (KU378551) / Vulpes_vulpes (KU378551)",
  "Vulpes_vulpes (KU378543) / Vulpes_vulpes (KU378543)",
  "Vulpes_vulpes (KU378544) / Vulpes_vulpes (KU378544)",
  "Vulpes_vulpes (KU378493) / Vulpes_vulpes (KU378493)",
  "Vulpes_vulpes (KU378554) / Vulpes_vulpes (KU378554)",
  "Vulpes_vulpes (KU378558) / Vulpes_vulpes (KU378558)",
  "Vulpes_vulpes (KU378553) / Vulpes_vulpes (KU378553)",
  "Vulpes_vulpes (KU378534) / Vulpes_vulpes (KU378534)",
  "Vulpes_vulpes (KU378519) / Vulpes_vulpes (KU378519)",
  "Vulpes_vulpes (KU378499) / Vulpes_vulpes (KU378499)",
  "Vulpes_vulpes (KU378496) / Vulpes_vulpes (KU378496)",
  "Vulpes_vulpes (KU378492) / Vulpes_vulpes (KU378492)",
  "Vulpes_vulpes (KU378563) / Vulpes_vulpes (KU378563)",
  "Vulpes_vulpes (KU378564) / Vulpes_vulpes (KU378564)",
  "Vulpes_vulpes (KU378559) / Vulpes_vulpes (KU378559)",
  "Vulpes_vulpes (KU378557) / Vulpes_vulpes (KU378557)",
  "Vulpes_vulpes (KU378556) / Vulpes_vulpes (KU378556)",
  "Vulpes_vulpes (KU378555) / Vulpes_vulpes (KU378555)",
  "Vulpes_vulpes (KU378539) / Vulpes_vulpes (KU378539)",
  "Vulpes_vulpes (KU378540) / Vulpes_vulpes (KU378540)",
  "Vulpes_vulpes (KU378538) / Vulpes_vulpes (KU378538)",
  "Vulpes_vulpes (KU378537) / Vulpes_vulpes (KU378537)",
  "Vulpes_vulpes (KU378524) / Vulpes_vulpes (KU378524)",
  "Vulpes_vulpes (KU378536) / Vulpes_vulpes (KU378536)",
  "Vulpes_vulpes (KU378535) / Vulpes_vulpes (KU378535)",
  "Vulpes_vulpes (KU378533) / Vulpes_vulpes (KU378533)",
  "Vulpes_vulpes (KU378530) / Vulpes_vulpes (KU378530)",
  "Vulpes_vulpes (KU378528) / Vulpes_vulpes (KU378528)",
  "Vulpes_vulpes (KU378504) / Vulpes_vulpes (KU378504)",
  "Vulpes_vulpes (KU378505) / Vulpes_vulpes (KU378505)",
  "Vulpes_vulpes (KU378503) / Vulpes_vulpes (KU378503)",
  "Vulpes_vulpes (KU378494) / Vulpes_vulpes (KU378494)",
  "Vulpes_vulpes (KU378497) / Vulpes_vulpes (KU378497)",
  "Vulpes_vulpes (KU378579) / Vulpes_vulpes (KU378579)",
  "Vulpes_vulpes (KU378576) / Vulpes_vulpes (KU378576)",
  "Vulpes_vulpes (KU378580) / Vulpes_vulpes (KU378580)",
  "Vulpes_vulpes (KU378578) / Vulpes_vulpes (KU378578)",
  "Vulpes_vulpes (KU378577) / Vulpes_vulpes (KU378577)",
  "Vulpes_vulpes (KU378575) / Vulpes_vulpes (KU378575)",
  "Vulpes_vulpes (KU378574) / Vulpes_vulpes (KU378574)",
  "Vulpes_vulpes (KU378573) / Vulpes_vulpes (KU378573)",
  "Vulpes_vulpes (KU378572) / Vulpes_vulpes (KU378572)",
  "Vulpes_vulpes (KU378571) / Vulpes_vulpes (KU378571)",
  "Vulpes_vulpes (KU378570) / Vulpes_vulpes (KU378570)",
  "Vulpes_vulpes (KU378569) / Vulpes_vulpes (KU378569)",
  "Vulpes_vulpes (KU378568) / Vulpes_vulpes (KU378568)",
  "Vulpes_vulpes (KU378565) / Vulpes_vulpes (KU378565)",
  "Vulpes_vulpes (KU378566) / Vulpes_vulpes (KU378566)",
  "Vulpes_vulpes (KU378567) / Vulpes_vulpes (KU378567)",
  "Vulpes_vulpes (KU378562) / Vulpes_vulpes (KU378562)",
  "Vulpes_vulpes (KU378561) / Vulpes_vulpes (KU378561)",
  "Vulpes_vulpes (KU378560) / Vulpes_vulpes (KU378560)",
  "Vulpes_vulpes (KU378515) / Vulpes_vulpes (KU378515)",
  "Vulpes_vulpes (KU378518) / Vulpes_vulpes (KU378518)",
  "Vulpes_vulpes (KU378552) / Vulpes_vulpes (KU378552)",
  "Vulpes_vulpes (KU378549) / Vulpes_vulpes (KU378549)",
  "Vulpes_vulpes (KU378548) / Vulpes_vulpes (KU378548)",
  "Vulpes_vulpes (KU378547) / Vulpes_vulpes (KU378547)",
  "Vulpes_vulpes (KU378546) / Vulpes_vulpes (KU378546)",
  "Vulpes_vulpes (KU378545) / Vulpes_vulpes (KU378545)",
  "Vulpes_vulpes (KU378542) / Vulpes_vulpes (KU378542)",
  "Vulpes_vulpes (KU378517) / Vulpes_vulpes (KU378517)",
  "Vulpes_vulpes (KU378541) / Vulpes_vulpes (KU378541)",
  "Vulpes_vulpes (KU378532) / Vulpes_vulpes (KU378532)",
  "Vulpes_vulpes (KU378529) / Vulpes_vulpes (KU378529)",
  "Vulpes_vulpes (KU378525) / Vulpes_vulpes (KU378525)",
  "Vulpes_vulpes (KU378526) / Vulpes_vulpes (KU378526)",
  "Vulpes_vulpes (KU378527) / Vulpes_vulpes (KU378527)",
  "Vulpes_vulpes (KU378523) / Vulpes_vulpes (KU378523)",
  "Vulpes_vulpes (KU378520) / Vulpes_vulpes (KU378520)",
  "Vulpes_vulpes (KU378522) / Vulpes_vulpes (KU378522)",
  "Vulpes_vulpes (KU378521) / Vulpes_vulpes (KU378521)",
  "Vulpes_vulpes (KU378516) / Vulpes_vulpes (KU378516)",
  "Vulpes_vulpes (KU378514) / Vulpes_vulpes (KU378514)",
  "Vulpes_vulpes (KU378513) / Vulpes_vulpes (KU378513)",
  "Vulpes_vulpes (KU378512) / Vulpes_vulpes (KU378512)",
  "Vulpes_vulpes (KU378511) / Vulpes_vulpes (KU378511)",
  "Vulpes_vulpes (KU378510) / Vulpes_vulpes (KU378510)",
  "Vulpes_vulpes (KU378508) / Vulpes_vulpes (KU378508)",
  "Vulpes_vulpes (KU378509) / Vulpes_vulpes (KU378509)",
  "Vulpes_vulpes (KU378507) / Vulpes_vulpes (KU378507)",
  "Vulpes_vulpes (KU378506) / Vulpes_vulpes (KU378506)",
  "Vulpes_vulpes (KU378501) / Vulpes_vulpes (KU378501)",
  "Vulpes_vulpes (KU378500) / Vulpes_vulpes (KU378500)",
  "Vulpes_vulpes (KU378502) / Vulpes_vulpes (KU378502)",
  "Vulpes_vulpes (KU378491) / Vulpes_vulpes (KU378491)",
  "Vulpes_vulpes (KU378498) / Vulpes_vulpes (KU378498)",
  "Vulpes_vulpes (KU378495) / Vulpes_vulpes (KU378495)",
  "Vulpes_vulpes (EF689062) / Vulpes_vulpes (EF689062)",
  "Vulpes_vulpes (EF689060) / Vulpes_vulpes (EF689060)",
  "Vulpes_vulpes (EF689058) / Vulpes_vulpes (EF689058)",
  "Vulpes_vulpes (AY928669) / Vulpes_vulpes (AY928669)",
  "Vulpes_vulpes (DQ498125) / Vulpes_vulpes (DQ498125)",
  "Vulpes_vulpes (JX013647) / Vulpes_vulpes (JX013647)",
  "Vulpes_vulpes (JN652603) / Vulpes_vulpes (JN652603)",
  "Vulpes_vulpes (DQ498127) / Vulpes_vulpes (DQ498127)",
  "Vulpes_vulpes (DQ498126) / Vulpes_vulpes (DQ498126)",
  "Vulpes_vulpes (JN652604) / Vulpes_vulpes (JN652604)",
  "Vulpes_vulpes (DQ498124) / Vulpes_vulpes (DQ498124)",
  "Vulpes_vulpes (JN652608) / Vulpes_vulpes (JN652608)",
  "Vulpes_vulpes (JQ003578) / Vulpes_vulpes (JQ003578)",
  "Vulpes_vulpes (JX013648) / Vulpes_vulpes (JX013648)",
  "Vulpes_vulpes (JX013646) / Vulpes_vulpes (JX013646)",
  "Vulpes_vulpes (JQ003577) / Vulpes_vulpes (JQ003577)",
  "Vulpes_vulpes (JN652618) / Vulpes_vulpes (JN652618)",
  "Vulpes_vulpes (JN652617) / Vulpes_vulpes (JN652617)",
  "Vulpes_vulpes (JN652615) / Vulpes_vulpes (JN652615)",
  "Vulpes_vulpes (JN652616) / Vulpes_vulpes (JN652616)",
  "Vulpes_vulpes (JN652614) / Vulpes_vulpes (JN652614)",
  "Vulpes_vulpes (JN652613) / Vulpes_vulpes (JN652613)",
  "Vulpes_vulpes (JN652612) / Vulpes_vulpes (JN652612)",
  "Vulpes_vulpes (JN652609) / Vulpes_vulpes (JN652609)",
  "Vulpes_vulpes (JN652611) / Vulpes_vulpes (JN652611)",
  "Vulpes_vulpes (JN652610) / Vulpes_vulpes (JN652610)",
  "Vulpes_vulpes (JN652607) / Vulpes_vulpes (JN652607)",
  "Vulpes_vulpes (JN652606) / Vulpes_vulpes (JN652606)",
  "Vulpes_vulpes (JN652605) / Vulpes_vulpes (JN652605)",
  "Vulpes_vulpes (KM657041) / Vulpes_vulpes (KM657041)",
  "Vulpes_vulpes (KM657042) / Vulpes_vulpes (KM657042)",
  "Vulpes_vulpes (KM657043) / Vulpes_vulpes (KM657043)",
  "Vulpes_vulpes (KM657039) / Vulpes_vulpes (KM657039)",
  "Vulpes_vulpes (KM657040) / Vulpes_vulpes (KM657040)",
  "Vulpes_vulpes (KM657038) / Vulpes_vulpes (KM657038)",
]
`;
